### PR TITLE
Async & cached loggers

### PR DESCRIPTION
**Pull request by Mykhailenko**

### CODE REVIEW

File: config/rm/settings.ini [hunk 1]
```diff
@@ -157,6 +157,20 @@ pa.rm.shutdown.kill.rt=true
...
+pa.log4j.async.appender.cache.enabled=false
+
+# Defines the buffer size used in asynchronous appenders
+pa.log4j.async.appender.buffer.size=10000
+
+# Defines the size of the thread pool of the AsynchFileAppender
+pa.log4j.async.appender.pool.size=1
+
+# Defines the AsynchFileAppender flush timeout
+pa.log4j.async.appender.flush.timeout=300
```
Comment (fviale) [inline config/rm/settings.ini]: 50ms seems better

File: rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/core/properties/PAResourceManagerProperties.java [hunk 2]
```diff
@@ -323,7 +323,32 @@
...
+
+    /**
+     * Defines the size of the thread pool of the AsynchFileAppender
+     */
+    LOG4J_ASYNC_APPENDER_POOL_SIZE("pa.log4j.async.appender.pool.size", PropertyType.INTEGER, "1"),
+
+    /**
+     * Defines the AsynchFileAppender flush timeout
+     */
+    LOG4J_ASYNC_APPENDER_FLUSH_TIMOUT("pa.log4j.async.appender.flush.timeout", PropertyType.INTEGER, "300");
```
Comment (fviale) [inline rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/core/properties/PAResourceManagerProperties.java]: 50ms seems better

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ServerJobAndTaskLogs.java [hunk 3]
```diff
@@ -209,7 +251,15 @@ private String readFile(File file) {
     void removeLogsDirectory() {
         String logsLocation = getLogsLocation();
         logger.info("Removing logs " + logsLocation);
-        FileUtils.removeDir(new File(logsLocation));
+        while (!org.apache.commons.io.FileUtils.deleteQuietly(new File(logsLocation))) {
+            logger.warn("Could not delete folder " + logsLocation + " retrying");
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ServerJobAndTaskLogs.java]: my mistake, InterruptedException should be catched outside the while

Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ServerJobAndTaskLogs.java]: ok

File: rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/core/properties/PAResourceManagerProperties.java (hunk 2 above)
Comment (Mykhailenko) [reply rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/core/properties/PAResourceManagerProperties.java]: how come 50 is in any sense better than 300?

**Comment by Mykhailenko** [comment]
Retest this please

File: rm/rm-server/src/main/java/org/ow2/proactive/utils/appenders/AsynchChachedFileAppender.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Override
+    public void append(String cacheKey, LoggingEvent event) {
+        try {
+            synchronized (queue) {
+                if (!appenderCache.containsKey(cacheKey)) {
+                    final RollingFileAppender appender = this.createAppender(cacheKey);
+                    appenderCache.put(cacheKey, appender);
+                }
+                //                appenderCache.computeIfAbsent(cacheKey, this::createAppender);
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/utils/appenders/AsynchChachedFileAppender.java]: can you put back the computeIfAbsent ?

File: rm/rm-server/src/main/java/org/ow2/proactive/utils/appenders/FileAppender.java [hunk 5]
```diff
@@ -79,55 +61,80 @@ public void append(LoggingEvent event) {
...
                 appender.setMaxFileSize(maxFileSize);
             }
-            appender.append(event);
-            appender.close();
-        } catch (IOException e) {
-            Logger.getRootLogger().error(e.getMessage(), e);
+        } catch (Exception e) {
+            Logger.getRootLogger()
+                  .error("Error when creating logger : " + cacheKey + " logging will be disabled for this context", e);
+            //            appender = new RollingFileAppender();
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/utils/appenders/FileAppender.java]: remove this commented out code, what about the exception ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ServerJobAndTaskLogs.java [hunk 6]
```diff
@@ -210,6 +253,15 @@ void removeLogsDirectory() {
         String logsLocation = getLogsLocation();
         logger.info("Removing logs " + logsLocation);
         FileUtils.removeDir(new File(logsLocation));
+        // this code breaks the tests
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ServerJobAndTaskLogs.java]: remove this code

**Pull request merged**